FurkannOzbek/application-hub
# [FEATURE] Implement Modal for Unsaved Changes Warning

# Pull Request Title

 [FEATURE] Implement Modal for Unsaved Changes Warning

---

## Description

This PR introduces a modal component that warns users about unsaved changes when navigating away from adding application page. The goal is to prevent accidental loss of data and enhance the user experience by providing a clear confirmation prompt before proceeding with a potentially disruptive action.

- Problem: Users were able to leave a page with unsaved changes without warning, leading to potential data loss.
- Impact: Prevents accidental loss of data, increasing user satisfaction and confidence in the system. 

---

## Changes Made
Frontend:

- Implemented a modal component to warn users of unsaved changes.
- Added state management to track if there are unsaved changes on the page.
- Updated navigation logic to trigger the modal when unsaved changes exist.
- Integrated with react-router to handle navigation actions.

Backend:
N/A – This change is frontend-only.

---

## Screenshots 
![ss5 ](https://github.com/user-attachments/assets/315b3b15-7618-490c-8877-33314f302795)
![ss6](https://github.com/user-attachments/assets/3871c0a8-7e53-46f2-9778-df58588889f3)

<img src="https://github.com/user-attachments/assets/e1f97122-b17c-4391-8fcd-d4e01396b709" alt="Description" width="300" />
<img src="https://github.com/user-attachments/assets/6cc126b8-c77d-43d4-9daa-104da9ea1c97" alt="Description" width="300" />

---

## Testing Steps

Frontend:

- Navigate to a add application page from dashboard after login.
- Make changes to the fields but do not save.
- Attempt to navigate away (e.g., another page).
- Verify that the modal appears with appropriate messaging.
- Confirm or cancel the action and ensure the behavior is correct.
